### ISSUE TITLE
Custom regexes from nerd-icons are not respected

### ISSUE THREAD

**Issue by bricka** [rainstormstudio/treemacs-nerd-icons#2]
I have made a custom entry in my `nerd-icons` configuration:

```emacs-lisp
(use-package nerd-icons
  :config
  (add-to-list 'nerd-icons-regexp-icon-alist '("^gradlew\\(.bat\\)?$" nerd-icons-sucicon "nf-seti-gradle" :face nerd-icons-silver)))
```

When I run `nerd-icons-icon-for-file`, I get this new icon:
![image](https://user-images.githubusercontent.com/318395/236428118-460af56d-d3d0-430d-8ecd-feaa6fb4bae4.png)

However, in my Treemacs, I still see the default icon:
![Screenshot 2023-05-05 at 11 54 46](https://user-images.githubusercontent.com/318395/236428287-6744eacd-7740-4a07-99aa-001c2997058f.png)

I think that the reason is that https://github.com/rainstormstudio/treemacs-nerd-icons/blob/main/treemacs-nerd-icons.el#L51 is only copying the `nerd-icons-extension-icon-alist` list, and not `nerd-icons-regexp-icon-alist`.

Thanks!

**Comment by rainstormstudio** [comment]
yes the thing is treemacs does not support regexp for extension recognition. I think the best you can do is to add it to `nerd-icons-extension-icon-alist` without using regexp.
